nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -153,6 +156,10 @@ let
           poolImported "${pool}" || poolImport "${pool}"  # Try one last time, e.g. to import a degraded pool.
         fi
         if poolImported "${pool}"; then
+        ${let cmd = concatMapStringsSep "\n" (elem: "clevis decrypt < /etc/clevis/${elem}.jwe | zfs load-key ${elem}") (filter (p: (elemAt (splitString "/" p) 0) == pool) clevisDatasets); in
+            optionalString (cmd != "") (cmd + " || ")}
```

Comment:
I just realized what this `||` is doing. Isn't that a little fragile? It relies on what follows to be a single bash command (which it is). But that's easily something someone could change without realizing what it does to this clevis bit.

Reply:
Hmm I see what you mean, what do you say I do `|| exit 0` instead ? The reason I just not to trigger a unit failure.

Reply:
Hm, actually, now that I think about it, what if only *some* encrypted datasets are decrypted with clevis? I mean I know that'd be weird, and maybe we can just ignore this usecase as a "why would you do that" kind of thing. In which case, `|| exit 0` seems like a fine solution.

Reply:
Yeah it needs a little bit refactor but we should for each clevisDevice try to unlock it using clevis or exit 0,
then in the rest of the code zfs will try to import every pool. If some have been unlocked already it will be happy and go to the next one, otherwise it will ask for the passphrase.

Reply:
Yeah not `|| exit 0` but `|| true`

Reply:
But I've done the change, each clevis decryption has its own independent bash line that can not have non zero return code.